tck/src/main/java/org/reactivestreams/tck/support/PublisherVerificationRules.java

```diff
@@ -11,46 +11,489 @@
+   * <li>make sure the {@link #required_createPublisher1MustProduceAStreamOfExactly1Element()} and {@link #required_createPublisher3MustProduceAStreamOfExactly3Elements()} tests pass.</li>
+   * </ul>
+   * @throws Throwable allow arbitrary exceptions to be thrown by the implementation of this test
+   */
   void required_spec102_maySignalLessThanRequestedAndTerminateSubscription() throws Throwable;
+  /**
+   * Asks for a short {@code Publisher} (i.e., length 10), repeatedly subscribes to this {@code Publisher}, requests items
+   * one by one and verifies the {@code Publisher} calls the {@code onXXX} methods non-overlappingly.
+   * <p>The test is not executed if {@link org.reactivestreams.tck.PublisherVerification#maxElementsFromPublisher()} is less than 10.
+   * <p>Note that this test is probabilistic, that is, may not capture any concurrent invocation in a {code Publisher} implementation.
+   * Note also that this test is sensitive to cases when a {@code request()} call in {@code onSubscribe()} triggers an asynchronous
+   * call to the other {@code onXXX} methods. In contrast, the test allows synchronous call chain of 
+   * {@code onSubscribe -> request -> onNext}.
+   * <p>If this test fails, the following could be checked within the {@code Publisher} implementation:
+   * <ul>
+   * <li>the {@code TestEnvironment} has large enough timeout specified in case the {@code Publisher} has some time-delay behavior,</li>
+   * <li>make sure the {@link #required_createPublisher1MustProduceAStreamOfExactly1Element()} and {@link #required_createPublisher3MustProduceAStreamOfExactly3Elements()} tests pass,</li>
+   * <li>if a {@code request()} call from {@code onSubscribe()} could trigger an asynchronous call to {@code onNext()} and if so, make sure
+   * such {@code request()} calls are deferred until the call to {@code onSubscribe()} returns normally.</li>
+   * </ul>
+   * @throws Throwable allow arbitrary exceptions to be thrown by the implementation of this test
+   */
   void stochastic_spec103_mustSignalOnMethodsSequentially() throws Throwable;
+  /**
+   * Verifies that if the call to {@code PublisherVerification.createErrorPublisher()} returns a non-null {@code Publisher},
+   * it calls {@code onSubscribe} exactly once followed by a single call to {@code onError()} without issuing any requests and otherwise
+   * not throwing any exception.
+   * <p>If this test fails, the following could be checked within the error {@code Publisher} implementation:
+   * <ul>
+   * <li>the {@code Publisher.subscribe(Subscriber)} method has actual implementation,</li>
+   * <li>in the {@code Publisher.subscribe(Subscriber)} method, if there is an upstream {@code Publisher},
+   * that {@code Publisher} is actually subscribed to,</li>
+   * <li>if the {@code Publisher} implementation is able to emit an {@code onError} without requests,</li> 
+   * <li>if the {@code Publisher} is non-empty as this test requires a {@code Publisher} without items.</li> 
```

Comment:
Indeed, some `Publisher` may want to emit items before emitting an `onError` thus in order to get to the error, requests must be issued. This is called a **delay-error** mode in RxJava and Reactor and is there as a choice for library users that want to process elements that were received successfully. When the TCK encounters such `Publisher`, this test times out because nobody asked for the items before the error.

I really don't want to argue about the value of this mode in contrast to the more victorian **fail-fast** mode certain "resilient" implementations are likely aiming at, thus warning the TCK user about trying to verify a `Publisher` that has some form of **delay-error** behavior.